app/controllers/users_controller.rb

```diff
@@ -0,0 +1,22 @@
+class UsersController < ApplicationController
+
+    def new
+        
+    end 
+
+    def create
+        user = User.new(user_params)
+        if user.save
+            redirect_to "/profile"
+        else
+            flash[:error] = user.errors.full_messages.to_sentence  
+            render :new
```

Comment:
This built in Rails error flash message will tell the user exactly what fields were not filled out  on the form.

Reply:
This looks great, a good idea to take the error messages from ActiveRecord instead of generating one for each case.

Reply:
Nice! I learned a new way to implement flash error messages.

Reply:
This flash message looks great. It seems like everyone has their own way of implementation.